client/starwhale/api/_impl/service/service.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+import os
+import typing as t
+import functools
+
+import pkg_resources
+from fastapi import FastAPI
+from pydantic import BaseModel
+from starlette.responses import FileResponse
+from starlette.staticfiles import StaticFiles
+
+from starwhale.base.models.base import SwBaseModel
+
+from .types import ServiceType
+
+STATIC_DIR_DEV = os.getenv("SW_SERVE_STATIC_DIR") or pkg_resources.resource_filename(
+    "starwhale", "web/ui"
+)
+
+
+class Query(BaseModel):
+    content: str
+
+
+class Api(SwBaseModel):
+    func: t.Callable
+    uri: str
+    inference_type: ServiceType
+
+    @staticmethod
+    def question_answering(func: t.Callable) -> t.Callable:
+        def inter(query: Query) -> str:
+            return func(query.content)  # type: ignore
+
+        return inter
+
+    def view_func(self, ins: t.Any = None) -> t.Callable:
+        func = self.func
+        if ins is not None:
+            func = functools.partial(func, ins)
+        return getattr(self, self.inference_type.value)(func)  # type: ignore
+
+
+class ServiceSpec(SwBaseModel):
+    title: t.Optional[str]
+    description: t.Optional[str]
+    version: str
+    apis: t.List[Api]
+
+
+class Service:
+    def __init__(self) -> None:
+        self.apis: t.Dict[str, Api] = {}
+        self.api_within_instance_map: t.Dict[str, t.Any] = {}
+
+    def api(self, inference_type: ServiceType) -> t.Any:
+        def decorator(func: t.Any) -> t.Any:
+            self.add_api(func, func.__name__, inference_type=inference_type)
```

Comment:
two func have the same name will cause overridding

Reply:
fixed